tests/test_009_layers.py

```diff
@@ -108,21 +107,28 @@ def test_set_layer_botm(plot=False):
 
 
 def test_insert_layer():
+    # download regis
     ds1 = get_regis_horstermeer()
+
     # just replace the 2nd layer by a new insertion
     layer = ds1.layer.data[1]
     new_layer = "test"
+    ds1_top3d = ds1["botm"] + nlmod.layers.calculate_thickness(ds1)
     ds2 = nlmod.layers.insert_layer(
-        ds1, "test", ds1["top"].loc[layer], ds1["botm"].loc[layer]
+        ds1, "test", ds1_top3d.loc[layer], ds1["botm"].loc[layer]
     )
+
+    # make sure the total thickness of the two models is equal (does not assert to True yet)
```

Comment:
Why is that... ?

Reply:
No idea yet... It is a difference of 0.125 if I sum the total thickness. I have no idea where this comes from.